src/bluehawk/actions/run.ts

```diff
@@ -0,0 +1,248 @@
+import { WithActionReporter } from "./ActionReporter";
+import { loadProjectPaths, snip, check, copy } from "../../bluehawk";
+import { ActionArgs } from "./ActionArgs";
+import { promises as fsp } from "fs";
+import { System } from "../io/System";
+import * as Path from "path";
+import YAML from "yaml";
+
+export interface ConfigArgs extends ActionArgs {
+  configPath?: string;
+}
+
+export interface Config {
+  commands: ConfigAction[];
+}
+
+export interface ConfigAction {
+  command: "snip" | "copy" | "check";
+  source: string;
+  destination: string;
+  ignore?: string[];
+  state?: string;
+  format?: "rst" | "md" | "docusaurus";
+  json?: boolean;
+}
+
+export const run = async (
+  args: WithActionReporter<ConfigArgs>
+): Promise<void> => {
+  const { configPath, reporter } = args;
+  const currentWorkingDirectory = Path.resolve(process.cwd());
+  const configFileName = "bluehawk.config.yaml";
+
+  let rootConfig: Config | undefined = undefined;
+  let rootConfigPath = configPath ? configPath : undefined;
+
+  // Get and parse root config file
+  const parseConfigFile = async (configFilePath: string) => {
+    const configFile = await fsp.readFile(configFilePath, "utf-8");
+    const parsedConfigFile: Config = YAML.parse(configFile);
+
+    return parsedConfigFile;
+  };
+
+  // Look for "bluehawk.config.yaml" in current working directory. If not
+  // there, traverse up the file path to system root.
+  const getRootConfigFile = async (
```

Comment:
Not sure we even want this functionality?
In what situations would you want to have bluehawk crawl up a directory?
I could definitely see it going down from where you run the command (e.g. run bluehawk in root of repo will execute bluehawk configs in subdirectories).
I fear going up would lead to a lot of unexpected behavior.